plugins/mac-capture/mac-window-capture.m

```diff
@@ -45,7 +45,8 @@ static inline void capture_frame(struct window_capture *wc)
 	uint64_t ts = os_gettime_ns();
 	CGImageRef img = get_image(wc);
 	if (!img)
-		return;
+		img = CGImageCreate(NULL, NULL, NULL, NULL, NULL, NULL, NULL,
+				    NULL, NULL, NULL, NULL);
```

Comment:
This PR does weird things it definitely shouldn't do. It somehow seems to switch the selected window in the settings to the wallpaper window (`[Dock] Desktop Picture - DefaultDesktop.heic`), not just show the wallpaper (which wouldn't be a good idea anyways) - hence it doesn't go back to the app when it's reopened. (Why it switches the windows is beyond be right now, but that doesn't really matter).
Also it spams the debugger with this each frame when img is null:
```
2021-12-26 14:13:31.903030+0100 obs[11209:303927] [com.obsproject.obs-studio] CGImageCreate: invalid image size: 0 x 0.
2021-12-26 14:13:31.904070+0100 obs[11209:303927] [com.obsproject.obs-studio] CGContextSetBlendMode: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
2021-12-26 14:13:31.904127+0100 obs[11209:303927] [com.obsproject.obs-studio] CGContextDrawImage: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
```

Instead of creating a new picture and going through the process of process of then drawing it etc (which also seems to cause other issues), I would recommend just sending an empty frame and then returning:
```suggestion
		obs_source_output_video(wc->source, NULL);
		return;
```

Reply:
@gxalpha I agree with that assessment, feel free to open a new PR for that if you still think this is necessary.